openlibrary/plugins/upstream/covers.py

```diff
@@ -71,8 +72,8 @@ def upload(self, key, i):
             upload_url = "http:" + upload_url
 
         try:
-            response = urllib.request.urlopen(upload_url, urllib.parse.urlencode(params))
-            out = response.read()
+            response = requests.post(upload_url+"?"+urllib.parse.urlencode(params), data=data)
+            out = response.content
         except urllib.error.HTTPError as e:
```

Comment:
```suggestion
            response = requests.post(upload_url+"?"+urllib.parse.urlencode(params), data=data)
            response.raise_for_status()
            out = response.content
        except requests.exceptions.HTTPError as e:
```

Reply:
requests will do parameter encoding if you just pass it a dict in the `params` arg e.g `requests.post(url, params=params, data=data)`

Reply:
Does `response = requests.post(upload_url, params=params, data=data)` work?

Reply:
Yes it does @cclauss 😄